pyvista/core/utilities/points.py

```diff
@@ -459,3 +464,156 @@ def vector_poly_data(orig, vec):
     pdata.GetPointData().SetActiveScalars(name)
 
     return pyvista.PolyData(pdata)
+
+
+def principal_axes(points: MatrixLike[float], *, return_std: bool = False):
+    """Compute the principal axes of a set of points.
+
+    Principal axes are orthonormal vectors that best fit a set of points. The axes
+    are also known as the principal components in Principal Component Analysis (PCA),
+    or the right singular vectors from the Singular Value Decomposition (SVD).
+
+    The axes are computed as the eigenvectors of the covariance matrix from the
+    mean-centered points, and are processed to ensure that they form a right-handed
+    coordinate frame.
+
+    The axes explain the total variance of the points. The first axis explains the
+    largest percentage of variance, followed by the second axis, followed again by
+    the third axis which explains the smallest percentage of variance.
+
+    The axes may be used to build an oriented bounding box or to align the points to
+    another set of axes (e.g. the world XYZ axes).
+
+    .. note::
+        The computed axes are not unique, and the sign of each axis direction can be
+        arbitrarily changed.
+
+    .. note::
+        This implementation creates a temporary array of the same size as the input
+        array, and is therefore not optimal in terms of its memory requirements.
+        A more memory-efficient computation may be supported in a future release.
+
+    .. versionadded:: 0.45.0
+
+    Parameters
+    ----------
+    points : MatrixLike[float]
+        Nx3 array of points.
+
+    return_std : bool, default: False
+        If ``True``, also returns the standard deviation of the points along each axis.
+        Standard deviation is computed as the square root of the eigenvalues of the
+        mean-centered covariance matrix, divided by the number of points.
+
+    Returns
+    -------
+    numpy.ndarray
+        3x3 orthonormal array with the principal axes as row vectors.
+
+    numpy.ndarray
+        Three-item array of the axis sizes.
+
+    Examples
+    --------
+    >>> import pyvista as pv
+    >>> import numpy as np
+    >>> rng = np.random.default_rng(seed=0)
```

Comment:
Might be better to protect end users with something like:
```suggestion
    >>> rng = np.random.default_rng(seed=0)  # only seeding for the example!
```
or just removing the seeding and allowing variation in the doctest results.